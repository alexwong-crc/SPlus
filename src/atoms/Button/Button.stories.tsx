import React from "react";
import Button from "./index";

export default { title: "Button" };

export const withText: React.SFC = () => <Button>Click Here</Button>;

export const withLoader: React.SFC = () => <Button loading>Click Here</Button>;
