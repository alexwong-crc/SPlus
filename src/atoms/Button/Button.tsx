import React from "react";

interface IProps {
  name: string;
}

export default (props: IProps) => <button>Hello {props.name}</button>;
