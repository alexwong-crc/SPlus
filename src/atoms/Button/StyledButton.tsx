import styled from "styled-components";

interface IStyledButton {
  type: "button" | "submit" | "reset";
  showText?: boolean;
}

const StyledButton = styled.button<IStyledButton>`
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  background-color: ${({ theme }): string => theme.action.main.code};
  color: ${({ theme }): string => theme.action.main.text};
  border: none;
  font-size: 0.8rem;
  position: relative;
  :active {
    background-color: ${({ theme }): string => theme.action.dark.code};
  }
  :hover {
    cursor: pointer;
  }
  :focus {
    outline: none;
  }
  :disabled {
    cursor: default;
    opacity: 0.6;
    background-color: ${({ theme }): string => theme.action.main.code};
  }
  span.text {
    opacity: ${({ showText }): number => (showText ? 1 : 0)};
  }
  span.icon {
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
  }
`;

export default StyledButton;
