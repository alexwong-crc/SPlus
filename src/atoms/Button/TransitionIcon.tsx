import React from "react";

const TransitionIcon: React.SFC = () => {
  return <div>hello</div>;
};

export default TransitionIcon;
