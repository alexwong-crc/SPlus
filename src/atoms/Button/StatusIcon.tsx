import React from "react";
import styled from "styled-components";
import { StatusProps } from "./Button";
import Icon from "../Icon";

interface IStatusIcon {
  status?: StatusProps;
}

type IconNames = "spinner" | "check" | "times";

const StyledIcon = styled.div<IStatusIcon>`
  position: absolute;
  transform: translateY(-200%);
  transition: 0.3s;
  &.slideInOut {
    animation-name: slideInOut;
    animation-duration: 1s;
  }

  @keyframes slideInOut {
    0% {
      transform: translateY(-200%);
    }
    25% {
      transform: translateY(0);
    }
    75% {
      transform: translateY(0);
    }
    100% {
      transform: translateY(-200%);
    }
  }
`;

const StatusIcon: React.SFC<IStatusIcon> = ({ status }: IStatusIcon) => {
  const renderIcon = (): IconNames => {
    switch (status) {
      case "success":
        return "check";
      case "failure":
        return "times";
      default:
        return "times";
    }
  };

  return (
    <>
      <StyledIcon
        className={
          status === "success" || status === "failure" ? "slideInOut" : ""
        }
      >
        <Icon icon={renderIcon()} transform="grow-5" />
      </StyledIcon>
      {status === "loading" ? (
        <Icon icon="spinner" spin transform="grow-5" />
      ) : null}
    </>
  );
};

export default StatusIcon;
