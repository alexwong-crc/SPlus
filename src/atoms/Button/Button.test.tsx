import React from "react";
import { shallow } from "enzyme";
import Button from "./index";

describe("Button", () => {
  it("should snapshot the button", () => {
    const component = shallow(<Button name="typescript test" />);
    expect(component).toMatchSnapshot();
  });
});
