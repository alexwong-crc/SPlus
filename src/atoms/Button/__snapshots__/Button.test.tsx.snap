// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should snapshot the button 1`] = `
<button>
  Hello 
  typescript test
</button>
`;
